domain/application/resource/store.go

```diff
@@ -4,64 +4,96 @@
+func (f FileResourceStore) Get(
+    ctx context.Context,
+    applicationUUID application.ID,
+    name string,
+) (io.ReadCloser, int64, error) {
+    path := f.storagePath(name, applicationUUID)
+    return f.objectStore.Get(ctx, path)
+}
+
+// Put the given resource in the object store.
+func (f FileResourceStore) Put(ctx context.Context, r io.Reader, resource Resource) error {
+    path := f.storagePath(resource.Name, resource.ApplicationID)
+    return f.objectStore.PutAndCheckHash(ctx, path, r, resource.Size, resource.Fingerprint.String())
+}
+
+// Remove the specified resource from the object store.
+func (f FileResourceStore) Remove(ctx context.Context, applicationUUID application.ID, name string) error {
+    path := f.storagePath(name, applicationUUID)
+    return f.objectStore.Remove(ctx, path)
 }
 
-func (f FileResourceStore) Get(ctx context.Context, path string) (io.ReadCloser, int64, error) {
-	return f.objectStore.Get(ctx, path)
+// storagePath returns the path at which the resource will be stored in the
+// object store. It is stored under the application name.
+func (f FileResourceStore) storagePath(name string, applicationUUID application.ID) string {
+    return path.Join("application-"+applicationUUID.String(), "resources", name)
 }
 
 // ResourceStoreFactory contains the information to provide the required
 // ResourceStore.
 type ResourceStoreFactory struct {
-	objectStore objectstore.ModelObjectStoreGetter
-	storeMap    map[resource.Type]ResourceStore
+    objectStore objectstore.ModelObjectStoreGetter
+    storeMap    map[resource.Type]ResourceStore
 }
 
 // NewResourceStoreFactory returns a factory which provides the appropriate
 // Resource Store for the resource type.
 func NewResourceStoreFactory(objectStore objectstore.ModelObjectStoreGetter) *ResourceStoreFactory {
-	return &ResourceStoreFactory{
-		objectStore: objectStore,
-		storeMap:    make(map[resource.Type]ResourceStore),
-	}
+    return &ResourceStoreFactory{
+        objectStore: objectStore,
+        storeMap:    make(map[resource.Type]ResourceStore),
+    }
 }
 
 // GetResourceStore returns the appropriate ResourceStore for the
 // given resource type.
 func (f *ResourceStoreFactory) GetResourceStore(ctx context.Context, t resource.Type) (ResourceStore, error) {
-	switch t {
-	case resource.TypeFile:
-		store, err := f.objectStore.GetObjectStore(ctx)
-		if err != nil {
-			return nil, err
-		}
-		return FileResourceStore{objectStore: store}, nil
-	default:
-		store, ok := f.storeMap[t]
-		if !ok {
-			return nil, fmt.Errorf("unknown resource type %q", t)
-		}
-		return store, nil
-	}
+    switch t {
+    case resource.TypeFile:
+        store, err := f.objectStore.GetObjectStore(ctx)
+        if err != nil {
+            return nil, err
+        }
+        return FileResourceStore{objectStore: store}, nil
+    default:
+        store, ok := f.storeMap[t]
+        if !ok {
+            return nil, fmt.Errorf("unknown resource type %q", t)
```

Comment:
**suggestion:** We should have specific error and use internal/errors

Reply:
@gfouillet That's a comment on the other PR I have up.

Reply:
Will rebase this as soon as thats updated.